### DISCUSSION TITLE
Why `<Outlet />`?

### DISCUSSION THREAD

**Discussion by bywo** [remix-run/react-router#8692] [Q&A]
I'm currently attempting upgrading a RR3 app to RR6. I've been waiting for nested route configuration with hooks, so thank you for all the hard work!

One difference I've encountered is that in RR3, the output of a child `Route` was passed as `children` to the parent `Route`'s element. In RR6, my parent `Route`'s element must render an `<Outlet />` instead of `children`.

As an example:
```jsx
// RR3
<Route path="/" component={PageLayout}>
  <IndexRoute component={Index} />
  <Route path="about" component={About} />
</Route>

// RR6
// Now, PageLayout must render an <Outlet /> instead of props.children
<Routes>
  <Route path="/" element={<PageLayout />}>
    <Route index component={<Index />} />
    <Route path="about" component={<About />} />
  </Route>
</Routes>
```

I'm curious as to the rationale behind the change. Thanks!

**Comment by grooting** [comment]
I wonder this too. Can someone give an answer?

**Comment by Pyrolistical** [comment]
I think it is a limitation of `element={<SomeComponent />} />`. `react-router` can't give it `children` as it has already been defined using JSX.

However, imo it ought to work if you use `Component={SomeComponent}`.
